lib/account.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+require 'time'
+
+module Bank
+  class Account
+    attr_reader :id, :balance, :date
+    def initialize(id, balance, date="nil")
+      @id = id
+      @balance = balance
+      @date = date
+      raise ArgumentError.new("balance must be >= 0") if balance < 0
+    end
+
+    def self.find(id1)
+      @accounts_array.each do |line|
+        return line if line.id == id1
+        raise ArgumentError.new("no valid id") if line.id != id1
+      end
+    end
+
+    def self.all
+      @accounts_array = []
```

Comment:
You probably want a class variable (two `@`s) here. What you'v got is an instance variable defined at the class level, which can act a little weird in some situations. Better to just stick with the class variable, at least for now.